src/activity_handlers.cpp

```diff
@@ -307,17 +307,38 @@ static bool check_butcher_cbm( const int roll )
 }
 
 static void butcher_cbm_item( const std::string &what, const tripoint &pos,
-                              const time_point &age, const int roll )
+                              const time_point &age, const int roll, const std::vector<std::string> flags,
+                              const std::vector<std::string> faults )
 {
     if( roll < 0 ) {
         return;
     }
     if( item::find_type( itype_id( what ) )->bionic.has_value() ) {
         item cbm( check_butcher_cbm( roll ) ? what : "burnt_out_bionic", age );
+        if( !flags.empty() ) {
```

Comment:
Don't check for emptiness, just iterate over the flags. Iteration is well-defined even when the container is empty.